FOR REFERENCE PURPOSES ONLY - Docs 6555 - Introduction to Connectors - Using Exchange to Locate Connectors, Templates, and Examples (#466)
mulesoft/docs-connectors: modules/ROOT/pages/introduction/use-exchange-to-locate-connectors.adoc

= Use Exchange to Locate Connectors, Templates, and Examples

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../../assets/images/

Use Anypoint Exchange (Exchange) to discover connectors, templates, and examples, and to incorporate them into your app. Exchange serves as a repository for connectors, templates, and examples provided both by MuleSoft and by other organizations. Searches in Exchange apply only to the Exchange organization you are viewing.

Search in the https://www.mulesoft.com/exchange/[Public Exchange] to access MuleSoft's public assets. If you search in a private organization's Exchange, you need access rights to that organization.

> For more information about searching in Exchange, see xref:release-notes::connector/to-find-info[Find Asset Information].

Reviewer: ```suggestion
For more information about searching in Exchange, see xref:exchange::to-find-info.adoc[Find Asset Information].
```

You can discover connectors by starting the search in Exchange or by accessing Exchange from within an Anypoint Flow Designer or Anypoint Studio project:

* Start in Exchange to view a list of all MuleSoft-provided connectors or a subset of these connectors. From there, you can view a connector in Flow Designer or download it to Studio.
* Access Exchange from within a Flow Designer or Studio project to locate a template applicable to your use case.

=== Discover all Connectors Provided by MuleSoft

To use Exchange as a starting point for discovering all or a subset of MuleSoft-provided connectors:

. https://www.mulesoft.com/exchange/[Access Exchange].
. Under *All assets* on the left side of the screen, click *Provided by MuleSoft*.
. Select *Connectors* in the drop-down list to the left of the search bar.
. To filter the list of connectors, add text to the search field. For example, if you enter `Salesforce`, the search returns all Salesforce connectors.
. Click on a connector to view its details.
. To download the connector for use in Studio, click *Download* from the right menu. You must be logged into Anypoint Platform to download a connector.
+

> The system downloads the connector as a `.jar` file. For instructions on adding this file to Studio, see <<import-downloaded-connector-studio, Import a Downloaded Connector into Studio>.

Reviewer: ```suggestion
The system downloads the connector as a `.jar` file. For instructions on adding this file to Studio, see <<import-downloaded-connector-studio, Import a Downloaded Connector into Studio>>.
```

> You can also add a connector directly to a Flow Designer or Studio project, as described later in this topic.

Author: ```suggestion
You can also add a connector directly to a Flow Designer or Studio project.
```

> === [[add-connector-to-flow-designer]]Add a Connector to a Flow Designer App

Author: ```suggestion
[[add-connector-to-flow-designer]]
=== Add a Connector to a Flow Designer App
```

> When you are working in a Flow Designer project, you can add a connector directly to the project:

Author: We should probably have wording to make it clear that you can only use Flow Designer for supported connectors. (Flow Designer doesn't support all connectors)

. Log into Anypoint Platform.

> . Open Design Center and create a new Mule app.

Author: I think we need to be really clear that they are actually creating the app in Design Center -> Flow Designer. Design Center is just an umbrella for API Designer and Flow Designer.

. In the *Let's Get Started wizard*, click *Go straight to canvas*, in the bottom-left corner.
. Click the on Trigger card.
. Select the connector.
. Continue designing the flow, as described in the User Guide for your connector.

=== [[import-downloaded-connector-studio]]Import a Downloaded Connector into a Studio

If you downloaded a connector from Exchange, you can import the downloaded `.jar` file into a Studio project:

. From Studio, select *Import* from the *File* menu.
. In the Import wizard, expand the *Anypoint Studio* folder.
. Select *Packaged mule application (.jar)*, and click *Next*.
. In the *File* field, optionally enter a name for the file.
. Click the ellipses (…​) button.
. Select the `.jar` file to import.
. Click *Open*.

> === [[add-connector-to-studio-project]]Add a Connector Directly to a Studio Project

Author: ```suggestion
[[add-connector-to-studio-project]]
=== Add a Connector Directly to a Studio Project
```

When you are working in a Studio project, you can add a connector directly to the project:

. From your Studio project, click the Exchange icon in the Studio taskbar:
+
image::introduction/introduction-exchange-icon.png[Exchange Icon]
+
. Log into Exchange using your Anypoint Platform username and password.
. Search for the connector.
. Select the connector and click *Add to project*.
. Follow the prompts to add the connector.

=== Next Step

Follow the directions in the connector's User Guide for configuring the connector in Design Center (Flow Designer) or Studio.

== Discover and Add Templates to Flow Designer or Studio

> A template is a pre-build project that you can use as a starting-point for applicable use cases. Many templates include one or more connector configurations.

Reviewer: ```suggestion
A template is a pre-built project that you can use as a starting point for applicable use cases. Many templates include one or more connector configurations.
```

> You can discover templates by starting in Exchange or by accessing Exchange from within Flow Designer or Studio:

Author: ```suggestion
Discover templates by starting in Exchange or by accessing Exchange from within Flow Designer or Studio:
```

> * Start in Exchange to view a list of all MuleSoft-provided templates or a subset of them. From there, you can view a template in Flow Designer or download it to Studio.

Author: ```suggestion
* Start in Exchange to view a list of all, or a subset of, MuleSoft-provided templates. From there, you can view a template in Flow Designer or download it to Studio.
```

* Access Exchange from within Flow Designer or Studio to locate a template that fits your use case.

When you locate a template in Exchange, the landing page of the associated connector contains:

* An overview of what the template does
* Prerequisites for using the template
* Instructions for configuring the template

> === Discover all MuleSoft Templates

Author: ```suggestion
=== Discover MuleSoft Templates
```

Author: Should we say instead "Discover Connector Templates" since this is connector documentation? Or "Discover Anypoint Connector Templates"

Reviewer 2: No. The templates have other things in them besides connectors, so they are not "connector templates."

> You can use Exchange as a starting point for discovering all MuleSoft-provided templates:

Author: ```suggestion
You can use Exchange as a starting point for discovering MuleSoft-provided templates:
```

. https://www.mulesoft.com/exchange/[Log into Exchange].
. Under *All assets* on the left side of the screen, click *Provided by MuleSoft*.
. Select *Templates* in the drop-down list to the left of the search bar.
. To filter the list of templates, add text to the search field. For example, if you enter `Salesforce template`, the search returns all templates that use Salesforce Connector.
. Click on a template to view its details.
+
If a template can be used in Flow Designer, it has an *Open flow* button in the top-right corner of the details page. This option opens the template in read-only mode in Flow Designer, which enables you to find out which connectors and operations are used, how data changes as it moves through the flow, and what configurations for connecting to your systems are already set up.
+

> For information about using a downloaded template in Flow Designer, https://docs.mulesoft.com/design-center/import-template[Create a Mule App Quickly from a Template].

Reviewer: ```suggestion
For information about using a downloaded template in Flow Designer, refer to xref:design-center::import-template.adoc[Create a Mule App Quickly from a Template].
```

+
All templates can be downloaded to Studio.
+

> . To download a template to Studio, click *Download* from the right menu. You must be logged into Anypoint Platform to download a template.

Author: Right menu? Menu on the right or something like that. Otherwise, it could be misunderstood as meaning the correct menu.

+
The system downloads the template as a `.jar` file. For instructions on adding this file to Studio, see <<import-downloaded-template-studio, Import a Downloaded Template into a Studio project>>.

> === [[import-downloaded-templates-studio]]Import a Downloaded Template to Studio

Reviewer: ```suggestion
=== [[import-downloaded-template-studio]]Import a Downloaded Template to Studio
```

> === [[import-downloaded-templates-studio]]Import a Downloaded Template to Studio

Author: ```suggestion
[[import-downloaded-templates-studio]]
=== Import a Downloaded Template to Studio
```

If you downloaded a template from Exchange, you can import the downloaded `.jar` file into Studio:

. From Studio, select *Import* from the *File* menu.
. In the Import wizard, expand the *Anypoint Studio* folder.
. Select *Packaged mule application (.jar)*, and click *Next*.
. In the *File* field, optionally enter a name for the file.
. Click the ellipses (…​) button.
. Select the `.jar` file to import.
. Click *Open*.
+
Studio imports the template as a new Studio project.

> === [[add-template-directly-studio]]Add a Template Directly to Studio

Author: ```suggestion
[[add-template-directly-studio]]
=== Add a Template Directly to Studio
```

When you are working in a Studio, you can import a template directly from Exchange:

. From Studio, click the Exchange icon in the Studio taskbar.
. Log into Exchange using your Anypoint Platform username and password:
+
image::introduction/introduction-exchange-icon.png[Exchange Icon]
+
. Search for the template, either directly, or by accessing the template link on the landing page of the associated connector.
. Select the template and click *Open*.
+
Studio imports the template as a new Studio project.

=== Next Step

Follow the instructions on the template's landing page in Exchange to use the template.

== Discover and Add Examples to Studio

An example is a Studio project that demonstrates how to handle a use case or solution. Many examples include one or more connector configurations. Once you locate an example, you can import it into Studio and then run it.

You can discover examples by starting in Exchange or by accessing Exchange from within a Studio project:

> * Start in Exchange to view a list of all MuleSoft-provided Examples or a subset of them. From there, you can view download it to Studio.

Author: ```suggestion
* Start in Exchange to view a list of all MuleSoft-provided examples. From there, you can download it to Studio.
```

* Access Exchange from Flow Designer or Studio to locate an example that fits your project's use case.

When you locate an example in Exchange, the landing page explains how to set up and run the example.

> === Discover All MuleSoft Examples

Author: ```suggestion
=== Discover MuleSoft Examples
```

Author: Same comments as before. Should we say "Anypoint Connector Examples" since this is connector documentation?

Reviewer 2: No. The many examples are broader than connectors. So I think saying "connector examples" implies that they are examples of connectors only.

> To use Exchange as a starting point for discovering all MuleSoft-provided examples:

Author: ```suggestion
To use Exchange as a starting point for discovering MuleSoft-provided examples:
```

. Log into Exchange with your Anypoint username and password.
. Under *All assets* on the left side of the screen, click *Provided by MuleSoft*.
. Select *Examples* in the drop-down list to the left of the search bar.
. To narrow down the list of examples, enter applicable text in the Search field. For example, if you enter `Salesforce Connector`, the search returns all examples that use the Salesforce Connector.
. Click on an example to view the example details.

> . To download the example for use in Studio, click *Download* from the right menu.

Author: menu on the right or similar...

+
The system downloads the file as a `.jar` file.

=== Import a Downloaded Example to Studio

If you downloaded an example from Exchange, you can import the downloaded `.jar` file into Studio to create the new project:

> . From Studio, select *Import* from the *File* menu.

Author: ```suggestion
. From Studio, select *File > Import* from the menu.
```

. In the Import wizard, expand the *Anypoint Studio* folder.
. Select *Packaged mule application (.jar)*, and click *Next*.
. In the *File* field, optionally enter a name for the file.
. Click the ellipses (…​) button.
. Select the `.jar` file to import.
. Click *Open*.
+

> Studio imports the template as a new Studio project.

Author: ```suggestion
Studio imports the example as a new Studio project.
```

=== Add an Example Directly to Studio

> When you are working in a Studio, you can import an example directly from Exchange:

Reviewer: ```suggestion
When you are working in Studio, you can import an example directly from Exchange:
```

. From Studio, click the Exchange icon in the Studio taskbar:
+
image::introduction/introduction-exchange-icon.png[Exchange Icon]
+
. Log into Exchange using your Anypoint Platform username and password.
. Search for the example.
. Select the example and click *Open*.
+

> Studio imports the template as a new Studio project.

Author: ```suggestion
Studio imports the example as a new Studio project.
```
